lib/Doctrine/DBAL/Driver/PDOSqlsrv/Connection.php

```diff
@@ -40,6 +40,22 @@ public function __construct($dsn, $user = null, $password = null, array $options
     /**
      * {@inheritDoc}
      */
+    public function lastInsertId($name = null)
+    {
+        if (null === $name) {
```

Comment:
That's why I asked. Probably needs a new issue though, where we validate that the passed in value is not empty or such.